tests/JustSaying.Extensions.DependencyInjection.AwsCore.Tests/ClientFactoryTests.cs

```diff
@@ -0,0 +1,106 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using JustSaying.AwsTools;
+using Amazon;
+
+
+[assembly: CollectionBehavior(DisableTestParallelization = true)]
+
+namespace TestProject1;
+
+
+public class ClientFactoryTests
+{
+    [Fact]
+    public void ShouldReadConfigFromFile()
+    {
+
+        var config = new ConfigurationBuilder()
+            .AddJsonFile("appsettings.json")
+            .Build();
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        var provider = serviceCollection.BuildServiceProvider();
+        var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
+        var snsClient = clientFactory.GetSnsClient(RegionEndpoint.EUWest1);
+        Assert.True(snsClient.Config.ServiceURL.Equals("http://test.test/") == true);
+        Assert.True(snsClient.Config.UseHttp);
+    }
+
+
+    [Fact]
+    public void ShouldReadFromDefaultProfile()
+    {
+        Environment.SetEnvironmentVariable("AWS_PROFILE", null);
+        var bin = AppContext.BaseDirectory;
+        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
+        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
+        var config = new ConfigurationBuilder()
+            .AddEnvironmentVariables()
```

Comment:
Rather than set environment variables and need to disable parallelism and reset the variables at the end of the tests (which also won't happen if it fails), use [an in-memory configuration source](https://learn.microsoft.com/dotnet/core/extensions/configuration-providers#memory-configuration-provider) instead.

Reply:
I have moved the needed env vars up to the constructor so they are only set once. SOme of the tests were doing the wrong thing. Hopefully they read better now. Fixed the Path issue

Reply:
Using an in-memory source would still be preferable as it means we aren't changing global process state in the tests.

Reply:
Hey yeah I tried that out and the AWS SDK didn't seem to work without those values being set as environment variables. I will dig into the code for the SDK (if I can find it) and check they don't just read direct from env vars.

Reply:
Cool - if that's the way it works (even though `IConfiguation` is passed to them) then that's fine to do that if there's no other way. If you do keep that though, you should make this class implement `IDisposable` and reset the environment variables in the `Dispose()` method so the state is reset between tests.

Reply:
Hiya! Have been following this with some interest, and thought I'd chime in, but don't want to step on any toes :-). 

I do wonder if it's worth adopting this package if it doesn't follow the rules of loading from configuration sources as is expected in dotnet. 
In my experience this kind of configuration management leads to really brittle test suites that are slow due to lack of parallelization and flaky as tests aren't properly isolated due to the reliance on environmental configuration. The just saying suite should always run in parallel :).

If it's an issue in the upstream library it might be worth raising a bug there and waiting on the fix?

Reply:
Thanks @gkinsman  and @martincostello for the input. I reviewed the tests again and it think the tests relying on env vars weren't adding any value. They were just testing that the AWS dotnet library does what it says it should. I think it seemed a bit pointless testing there code for them here so removed it in the end. This has enabled just using an in memory collection.